frontends/parsers/p4/p4parser.ypp

```diff
@@ -1283,6 +1290,49 @@ statementOrDeclaration
     | instantiation            { $$ = $1; }
     ;
 
+forStatement
+    : FOR "(" forInitStatements ";"
+              expression ";"
+	      forUpdateStatements ")"
+      statement
+    { $$ = $9; }   // TODO: temporary placeholder code to avoid bison warning
+    | FOR "(" typeRef name IN forCollectionExpr ")"
+      statement
+    { $$ = $8; }   // TODO: temporary placeholder code to avoid bison warning
+    ;
+
+forInitStatements
+    : %empty
+    | forInitStatementsNonEmpty
+    ;
+
+forInitStatementsNonEmpty
+    : declOrAssignmentOrMethodCallStatement
+    | declOrAssignmentOrMethodCallStatement "," forInitStatementsNonEmpty
+    ;
+
+declOrAssignmentOrMethodCallStatement
+    : variableDeclarationWithoutSemicolon
+    | assignmentOrMethodCallStatementWithoutSemicolon
+    ;
+
+forUpdateStatements
+    : %empty
+    | forUpdateStatementsNonEmpty
+    ;
+
+forUpdateStatementsNonEmpty
+    : assignmentOrMethodCallStatementWithoutSemicolon
+    | assignmentOrMethodCallStatementWithoutSemicolon ","
+      forUpdateStatementsNonEmpty
+    ;
+
+forCollectionExpr
+    : expression
+    | expression ".." expression
+    | typeRef
```

Comment:
What is a `typeRef` supposed to indicate here?  Creating a temp of the type and then iterating over it?